webapp/src/main/java/org/apache/atlas/web/rest/FeatureFlagREST.java

```diff
@@ -0,0 +1,386 @@
+            }
+
+            if (StringUtils.isBlank(key)) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Feature flag key cannot be empty");
+            }
+
+            if (!FeatureFlag.isValidFlag(key)) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, 
+                    "Invalid feature flag key: " + key + ". Valid keys are: " + Arrays.toString(FeatureFlag.getAllKeys()));
+            }
+
+            FeatureFlag flag = FeatureFlag.fromKey(key);
+            String currentValue = FeatureFlagStore.getFlag(key);
+            
+            FeatureFlagInfo flagInfo = new FeatureFlagInfo();
+            flagInfo.setKey(key);
+            flagInfo.setCurrentValue(currentValue);
+            flagInfo.setDefaultValue(String.valueOf(flag.getDefaultValue()));
+            flagInfo.setTimestamp(new Date());
+
+            return flagInfo;
+
+        } finally {
+            AtlasPerfTracer.log(perf);
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("<== FeatureFlagREST.getFeatureFlag({})", key);
+            }
+        }
+    }
+
+    /**
+     * Update or set a feature flag value.
+     * 
+     * @param key Feature flag key
+     * @param request Update request containing the new value
+     * @param servletRequest HTTP servlet request
+     * @return FeatureFlagResponse indicating success
+     * @throws AtlasBaseException if flag is invalid or operation fails
+     */
+    @PUT
+    @Path("{key}")
+    @Timed
+    public FeatureFlagResponse updateFeatureFlag(@PathParam("key") String key,
+                                               FeatureFlagUpdateRequest request,
+                                               @Context HttpServletRequest servletRequest) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("==> FeatureFlagREST.updateFeatureFlag({}, {})", key, request != null ? request.getValue() : "null");
+        }
+
+        AtlasPerfTracer perf = null;
+        try {
+            if (AtlasPerfTracer.isPerfTraceEnabled(PERF_LOG)) {
+                perf = AtlasPerfTracer.getPerfTracer(PERF_LOG, "FeatureFlagREST.updateFeatureFlag(" + key + ")");
+            }
+
+            if (StringUtils.isBlank(key)) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Feature flag key cannot be empty");
+            }
+
+            if (request == null || StringUtils.isBlank(request.getValue())) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Feature flag value cannot be empty");
+            }
+
+            if (!FeatureFlag.isValidFlag(key)) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, 
+                    "Invalid feature flag key: " + key + ". Valid keys are: " + Arrays.toString(FeatureFlag.getAllKeys()));
+            }
+
+            // Set the feature flag
+            FeatureFlagStore.setFlag(key, request.getValue());
```

Comment:
Since this says `PUT`, I think we should always check GET for the key & also throw an exception in case flag was not present ?

Reply:
I missed that we do not have POST, ignore this one